### ISSUE TITLE
"kvazaar.exe" more verbose --version info

### ISSUE THREAD

**Issue by Vangelis66** [ultravideo/kvazaar#128]
(This is an **enhancement** request, not an issue as such!)

Hello - thanks for your ongoing efforts!
I am using the most excellent [media-autobuild_suite](https://github.com/jb-alvarado/media-autobuild_suite) script 
under Windows Vista SP2 32bit to compile x86 
standalone "kvazaar.exe" binaries, so this is where 
I posted my request at first: 

https://github.com/jb-alvarado/media-autobuild_suite/issues/297#issue-139375145 

but was directed here by its code maintainer...

Please do read the initial report there, 
it'd be counter productive to also copy-paste it here...

I have read https://github.com/ultravideo/kvazaar/issues/123 already that implements `--version`, 
what I am kindly asking now is for `--version` to be able 
to print something more informative, like:

`kvazaar-0.8.3-master-g1671725-20160308-i686-static`

given the number of frequent updates...

 This would make "kvazaar.exe" very similar 
to other encoders like "x264.exe" & "x265.exe", 
which by default show git(hg)-version & commit hash, 
thus making it easy to discern the source code 
they are built from....

Many thanks in anticipation...
Regards.

**Comment by Venti-** [comment]
I guess that would be nice to have. Not exactly sure how to do that, such that it would also work with configure being run outside the git folder, but passing the result of git describe to the compiler should be easy enough.

**Comment by Vangelis66** [comment]
@Venti wrote:

> I guess that would be nice to have.

Well, I am happy you feel positive about it... :-)

> Not exactly sure how to do that, such that it would also work 
> with configure being run outside the git folder, 
> but passing the result of git describe to the compiler 
> should be easy enough.

I sense you'll make good use of @wiiaboo 's tip:
https://github.com/jb-alvarado/media-autobuild_suite/issues/297#issuecomment-194029645

... Off to bed now, it's getting 
really late in my timezone!

Big thanks and kind regards.

**Comment by aryla** [comment]
I changed the displayed version to the output of git describe in 54f08f2bdb5c0eb2696f2ee5f30d72f6a2bf3b0f. Now it looks like `v0.8.3-47-g54f08f2[-dirty]`. The commit date and branch are not displayed yet.

**Comment by wiiaboo** [comment]
@aryla  The leading `v` should probably be stripped or stuff that depended on the old version string will fail with that commit (like FFmpeg's configure).

**Comment by Vangelis66** [comment]
@aryla wrote: 

> I changed the displayed version to the output of 
> git describe in https://github.com/ultravideo/kvazaar/commit/54f08f2bdb5c0eb2696f2ee5f30d72f6a2bf3b0f. 
> Now it looks like `v0.8.3-47-g54f08f2`. 

Thank you for your work so far. I compiled a fresh "kvazaar.exe",
which indeed now reports: 

`Kvazaar v0.8.3-52-g47a9f1a`
`Kvazaar license: LGPL version 2`

> The commit date and branch are not displayed yet.

So does this mean you are still optimistic 
it can be implemented in the near future?

However, @wiiaboo is right; compilation of 
ffmpeg.exe fails, because ffmpeg's configure fails :-( 
The following are the last lines of ffmpeg's config.log: 

`check_pkg_config kvazaar >= 0.8.1 kvazaar.h kvz_api_get`
`pkg-config --exists --print-errors kvazaar >= 0.8.1`
`Requested 'kvazaar >= 0.8.1' but version of libkvazaar is v0.8.3`
`ERROR: kvazaar not found using pkg-config`

How I really hate it when trying to improve one thing 
breaks other things in a broader level!
I feel guilty now since I was the one that started this...
I honestly hope it can be fixed soon!

Thanks in advance!

**Comment by Venti-** [comment]
Don't feel bad about encouraging change. When things change they sometimes break, but they also sometimes improve. It's in media-autobuild suites nature to always be broken because it's grabbing so many unstable constantly changing versions, which is also why this was caught and fixed so quickly by @wiiaboo. This was clearly our fault for not having automated tests for the pkgconfig interface, but now we have a reason to add those tests.

**Comment by Vangelis66** [comment]
> Don't feel bad about encouraging change

Actually, I do not feel bad about proposing what I consider as 
an enhancement, I do feel guilty though about the extra work 
caused to @wiiaboo today as a consequence...
Especially when the whole idea was that "_it'd be easier for everyone_" 
if my original request was taken care of here...

I think a famous song title is appropriate here 
(What a Diff'rence a Day Makes), paraphrased as 
**What a Diff'rence a "v" Makes**, but just for the 
record, I hadn't asked for that "v" to be included 
in my initial request...

> It's in media-autobuild suite's nature to **always be broken**

To be fair, I would reword the above to:  

 _It's in media-autobuild_suite's nature to break from time to time_

which, judging from experience, is not that often!

> but now we have a reason to add those tests.

Indeed, that would be an improvement, as an 
immediate result of today's mishaps... 

All's well that ends well; PR was merged and the 
suite repatched to accommodate...

>> The commit date and branch are not displayed yet.

You (or @aryla) have not indicated your intentions 
regarding this; I'll leave this issue open till then; 
if you deem it unnecessary or too difficult to implement, 
then say so and close this yourself...

All the best to the team, 
regards

**Comment by wiiaboo** [comment]
> which, judging from experience, is not that often!

I can almost tell every time @Selur runs it since there's always something wrong, so it's broken enough. It depends whether people use just a few extra stuff or a lot. Less stuff being build, less likely it is to break.

I deferred the question of adding more verbose version to the developers because they usually know better what to do with their project than me. If I patch enhancements to active projects it's more work for me to maintain them and for them if those patched versions show up bugged and differ too much in behavior. In this case it was a relatively low effort thing to add.

> The commit date and branch are not displayed yet.

Branch should always be master and commit date is less useful than commit hash, imo. It shouldn't be necessary to have those. Both can be inferred from the commit hash too unless there's `git am`'d patches applied.

**Comment by Vangelis66** [comment]
@wiiaboo wrote:

> Branch should always be **master** 

Indeed it should, so yes, one (and myself) can live without it!
But certain apps do include it in their version output; 
for example MP4box.exe compiled by GPAC themselves 
([Latest available win32 nightly binaries](http://download.tsi.telecom-paristech.fr/gpac/latest_builds/win32/)) does include it: 

`MP4Box - GPAC version 0.6.0-rev0-gb5e51f8-master`
`GPAC Copyright (c) Telecom ParisTech 2000-2012`
`GPAC Configuration: (static configuration file)`
`Features:` 
`GPAC_MEMORY_TRACKING`
`GPAC_HAS_SSL`
`GPAC_HAS_SPIDERMONKEY`
`GPAC_HAS_JPEG`
`GPAC_HAS_PNG`

> commit date is less useful than commit hash 
> (snip)
> It shouldn't be necessary to have those
> (snip)
> Both can be inferred from the commit hash

Of course I do agree that **commit hash** is the most 
important piece of information and we have that now, 
many thanks; if commit push date was also appended,
it would just be an extra benefit/luxury that would easily 
allow one to tell apart more recent code base between 
different compiles (that is without having to resort to 
git functions or the online repo...).
As stated already, I have left the final decision 
upon the Kvazaar devs themselves!

**Comment by aryla** [comment]
I added the commit date to the output in 2fcd677ba978f09012367090fa45401d513e269a.

Displaying the branch doesn't seem very useful to me so I think we'll leave it out.

**Comment by Vangelis66** [comment]
@aryla 

Yes, huge thanks! 
Freshly compiled binary now reports: 
```
Kvazaar v0.8.3-67-g6f431e5 2016-03-14
Kvazaar license: LGPL version 2
```
(**v0.8.3-67-g6f431e5-20160314** was what I 
was hoping for, but your way is equally good...)

In the words of Mr Spock:
_"Live long and prosper"_
(this goes to all the Kvazaar team!)

Many thanks indeed.
